src/constructs/core/stack.ts

```diff
@@ -138,6 +130,24 @@ export class GuStack extends Stack implements StackStageIdentity, GuMigratingSta
 
       this.tryAddRepositoryTag();
     }
+
+    const { node } = this;
+
+    node.addValidation({
```

Comment:
The validation step happens just before synthesis:

![image](https://user-images.githubusercontent.com/836140/168233942-d65d64cc-6579-4b8f-8148-a74a892160ad.png)

See:
  - [App lifecycle](https://docs.aws.amazon.com/cdk/v2/guide/apps.html#lifecycle)
  - The [`constructs` docs](https://github.com/aws/constructs/blob/bd0d7c30c794c9198bd7e866362cd4bc32eeed91/API.md#addvalidationvalidation-)